### PR TITLE
Delete ip range

### PR DESCRIPTION
**Pull request by pandeys**
VP 1446: [VCD-CLI] Delete an IP range of a subnet in external network

[VCD-CLI] Delete an IP range of a subnet in external network

This CLN contains code fro removing an IP range of a subnet in an
external network and it's test cases.
User can usel this command as:

vcd network external remove-ip-range <external_network_name>
--g <gateway-ip> --ip-range <startaddress-endaddress>

Testing done:
Test case for remove ip range is added to a test class,
extnet_tests.py
All tests in extnet_tests passed.

---
This change is [Reviewable](https://reviewable.io/reviews/vmware/vcd-cli/290)

### CODE REVIEW

File: system_tests/extnet_tests.py [hunk 1]
```diff
@@ -223,22 +198,61 @@ def test_0040_attach_port_group(self):
...
+            args=[
+                'detach-port-group', self._name, '--vc-name', vc_name,
+                '--port-group', pg_name])
+        self.assertEqual(0, result.exit_code)
+
+    def test_0050_add_ip_range(self):
+        """Add an IP range to a subnet in an external network.
+
+        Invoke the command 'external add-ip-range' in network.
+        """
```
Comment (kousgy123) [inline system_tests/extnet_tests.py]: Any reason "def test_0033_add_ip_range" is moved to test_0050_add_ip_range" ?

Any reason "def test_0031_modify_ip_range(self)" is move to "def test_0051_modify_ip_range(self):" ?

Why  "def test_0052_remove_ip_range(self)" can't be kept as "def test_0034_remove_ip_range(self)" ?

Comment (pandeys) [reply system_tests/extnet_tests.py]: It is done for grouping similar test cases like add and enable subnet start from 30 (we have buffer numbers, If any new test cases added in future related to subnet). Similarly ip_range related starting from 50.

File: system_tests/extnet_tests.py [hunk 2]
```diff
@@ -223,22 +198,61 @@ def test_0040_attach_port_group(self):
...
-                args=[
-                    'detach-port-group', self._name, '--vc-name', vc_name,
-                    '--port-group', pg_name])
-            self.assertEqual(0, result.exit_code)
+    def test_0041_detach_port_group(self):
+        """Detach port group from an external network.
+        Invoke the command 'external detach-port-group' in network.
+        """
+        ExtNetTest._client = Environment.get_sys_admin_client()
+        #       platform = Platform(self.client)
```
Comment (guptaankit52) [inline system_tests/extnet_tests.py]: Please remove this comment

File: vcd_cli/network.py [hunk 3]
```diff
@@ -765,6 +770,38 @@ def modify_ip_range_external_network(ctx, name, gateway_ip, ip_range,
...
+    metavar='<ip>',
+    help='gateway ip of the subnet')
+@click.option(
+    '-i',
+    '--ip-range',
+    'ip_range',
+    required=True,
+    metavar='<ip>',
+    help='ip range in StartAddress-EndAddress format')
+def remove_ip_range_external_network(ctx, name, gateway_ip, ip_range):
```
Comment (guptaankit52) [inline vcd_cli/network.py]: I feel We should provide user do delete multiple ip ranges, as from ui we can do it the samw and even in gateway we are doing the same?, please provide your input

Comment (guptaankit52) [reply vcd_cli/network.py]: I just checked even confluence page says the same for removing ip range:

https://confluence.eng.vmware.com/pages/viewpage.action?spaceKey=VCD&title=Design+of+External+Network+commands+for+vcd-cli

**Pull request merged**